src/commands/set/state-backend.ts

```diff
@@ -17,13 +17,25 @@ const SetStateBackendCommand = BaseCommand()
   .option('--creds <creds:string>', 'A key value pair of credentials to use for the provider', { collect: true })
   .action(set_state_backend);
 
+function getSecretProviderTypes(): string[] {
+  const results = [];
+  for (const [name, providerType] of Object.entries(SupportedProviders)) {
+    const provider = new providerType('test', {} as any, {} as any, {});
```

Comment:
The massive amount of empty inputs feels a bit weird. I know we don't have values for the credentials, but it looks like you can definitely pass in the providerStore to the `getSecretProviderTypes()` method from the command_helper. That would make me feel a bit better about the aliasing since the credentials is a simpler TS type.

Reply:
It does not need it though. We are just using this to check if the secret resource exists on the object, but we only know that if we instantiate it. No actual code from the class ever gets called from this. The intent here is to make it clear that this is just for analysis and not usable.